## Solutions to auto-diagnostic test 3
Exercises By L.C. Félix-Herrán, Tec.

### 3) 
Given $$ G(s) = \frac{3s\mathrm{e}^{-2s}}{s+1}, $$
Determine the discretization of the system assuming that it is preceeded by a hold and proceeded by an ideal sampler for the following cases

- **a)** ZOH and sampling period $h=0.5$
- **b)** ZOH and sampling period $h=0.3$
- **c)** FOH and sampling  period $h=0.5$

### Solutions
#### b)
The delay $\tau$ in the process is 2, so we have $$ \tau = 6h + mh$$ with $h=0.3$ and $ m = 2/3$. Let $y_0(t)$ be the step-response of the system without the delay, and $y_1(t)$ be the step-response of the system for a delay of $6h$. We get
\begin{align}
y_0(t) &= \mathcal{L}^{-1} \{ \frac{3s}{s+1} \cdot \frac{1}{s} \} = 3\mathrm{e}^{-t}u_H(t)\\
y_1(t) &= y_0(t-6h)\\
y(t) &= y_1(t-mh) = y_0(t-6h-mh)
\end{align}
Applying the z-transform to the sampled signal $y_1(kh)$ gives
$$ Y_1(z) = z^{-6} Y_0(z) = z^{-6} \frac{3z}{z-\mathrm{e}^{-h}}. $$
In order to obtain the z-transform of the actual step-response $y(t)$ we need to use the modified z-transform (See the textbook Fernández y Ramírez, ch 4.6).
\begin{align}
 Y(z, m) &= \sum_{k=0}^\infty y_1(kh+mh)z^{-k} = \sum_{k=0}^\infty 3\mathrm{e}^{-kh-mh}z^{-k}\\
 &= \mathrm{e}^{-mh} \sum_{k=0}^\infty 3\mathrm{e}^{-kh}z^{-k} = \mathrm{e}^{-mh} Y_1(z)\\
 &= \mathrm{e}^{-mh} \frac{3}{z^5(z-\mathrm{e}^{-h})}
 \end{align}

In [9]:
import numpy as np
h = 0.3
m = 2.0/3.0
emh = np.exp(-m*h)
eh = np.exp(-h)
print('e^{-mh}=%s' %(emh))
print('e^{-h}=%s' %(eh))
3*emh
np.exp(-0.33*0.3)

e^{-mh}=0.8187307530779818
e^{-h}=0.7408182206817179


0.9057427080235485

With numerical values
$$ Y(z,m=2/3) = 0.82 \frac{3}{z^5(z-0.74)}$$
All that is left is to divide by the z-transform of the step signal
\begin{align}
H(z) &= \frac{Y(z)}{U(z)} = \frac{z-1}{z}\frac{3\cdot 0.82}{z^5(z-0.74)}\\
&= \frac{2.46(z-1)}{z^6(z-0.74)}
\end{align}